Descent3/tests/porting-tests.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Descent 3
+ * Copyright (C) 2024 Descent Developers
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <gtest/gtest.h>
+
+#include "osiris_predefs.h"
+#include "osiris_share.h"
+#include "player_external_struct.h"
+
+// This is copy of actual Osiris_CreateGameChecksum(void) from OsirisLoadandBind.cpp
+uint Osiris_CreateGameChecksumTest() {
```

Comment:
Is there a reason Osiris_CreateGameChecksumTest can't actually call Osiris_CreateGameChecksum? you'd have to call Osiris_CreateModuleInitStruct first. seems like it would be better to test the actual code instead of something almost but not exactly the running code.

Reply:
Yeah very good point @jcoby, a unit test should be performing the check against the actual code used in game, not a copy. Then if someone breaks the real code, the test is still going to pass no matter what.

Reply:
To import Osiris_CreateGameChecksum() I need somehow import nearly half of project's code in order to just link it. Here goes cfile, mem, debug, sound components etc... For now I just wrote functions that reproduce Osiris_CreateGameChecksum() logic.
I hope in future code would be isolated enough to import Osiris_CreateGameChecksum() itself.

Reply:
Yeah something is odd with the osiris includes, we'll need to clean that